pkg/serverless/protocol.go

```diff
@@ -387,3 +394,15 @@ func (f *Flush) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}()
 
 }
+
+func areEnhancedMetricsEnabled() bool {
+	priorityEnv := "serverless.enhanced_metrics"
+	nonPriorityEnv := "enhanced_metrics"
```

Comment:
I think this approach of allowing either environment variable could make sense. I think we should consider the pros and cons of which one should be prioritized.

I do think that in terms of documentation, we probably want to document only the `DD_ENHANCED_METRICS` environment variable:

- Most customers will likely prefer to configure the extension using environment variables rather than a config file
- `DD_ENHANCED_METRICS` will enable/disable enhanced metrics in both the Lambda Library and the Extension, whereas if they used `DD_SERVERLESS_ENHANCED_METRICS`, it would only affect the extension and they would need to add `DD_ENHANCED_METRICS` anyway.

Because of that, maybe we should prioritize `DD_ENHANCED_METRICS`? And then fall back to `DD_SERVERLESS_ENHANCED_METRICS` if that is present in the config file?

Another option would be to ditch this logic of two different environment variables and just support `DD_ENHANCED_METRICS`. The only downside is that the config file would have to look like this:

```
enhanced_metrics: true
serverless:
     logs_enabled: true
```

However, most customers won't use the config file anyway -- and if they do, it won't apply to the Library, so it's not the best way to change this setting.

Sorry for all of the debating and back-and-forth around how to best handle this -- I do think it makes sense to think this through very carefully because once we add these environment variables/config options, they cannot be easily changed or removed.

Reply:
Thanks for raising this flag, this is indeed an important decision.
IMO we should 
- leave the two env variables because some customers are used to DD_ENHANCED_METRICS but we also have consistency in the config file 
- set DD_SERVERLESS_ENHANCED_METRICS with a higher priority than DD_ENHANCED_METRICS since I agree that not a lot of customers will use the configuration file but if they do, I think they might not expect any override be applied afterwards
Does it make sense ?

Reply:
Let's go with only one env variable